app/models/user.rb

```diff
@@ -28,4 +33,10 @@ def self.create_with_omniauth(auth)
       user.email = auth["info"]["email"]
     end
   end
+
+  def send_confirmation
+    generate_token(:confirmation_token)
+    self.confirmation_sent_at = Time.now
+    UserMailer.email_confirmation(self.confirmation_token).deliver_later
```

Comment:
so considering what i said above about the user_email here you should send `self` instead of the token.